reporter/src/main/kotlin/reporters/AttributionDocumentPdfReporter.kt

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (C) 2020 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.reporter.reporters
+
+import java.io.File
+import java.io.OutputStream
+import java.net.URL
+import java.net.URLClassLoader
+
+import org.eclipse.sw360.antenna.attribution.document.core.AttributionDocumentGeneratorImpl
+import org.eclipse.sw360.antenna.attribution.document.core.DocumentValues
+import org.eclipse.sw360.antenna.attribution.document.core.model.LicenseInfo
+
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.OrtResult
+import org.ossreviewtoolkit.model.utils.PackageConfigurationProvider
+import org.ossreviewtoolkit.model.utils.getDetectedLicensesWithCopyrights
+import org.ossreviewtoolkit.reporter.LicenseTextProvider
+import org.ossreviewtoolkit.reporter.Reporter
+import org.ossreviewtoolkit.reporter.ReporterInput
+import org.ossreviewtoolkit.reporter.utils.AttributionDocumentPdfModel
+import org.ossreviewtoolkit.spdx.SpdxLicense
+import org.ossreviewtoolkit.utils.ProcessedDeclaredLicense
+import org.ossreviewtoolkit.utils.safeDeleteRecursively
+import org.ossreviewtoolkit.utils.toHexString
+
+private const val DEFAULT_TEMPLATE_ID = "basic-pdf-template"
+private const val TEMPLATE_ID = "template.id"
+private const val TEMPLATE_PATH = "template.path"
+
+class AttributionDocumentPdfReporter : Reporter {
+    override val reporterName = "AttributionDocumentPDF"
+    override val defaultFilename = "attribution-document.pdf"
+
+    override fun generateReport(outputStream: OutputStream, input: ReporterInput, options: Map<String, String>) {
+        val ortResult = input.ortResult
+        val artifacts = ortResult.getPackages().map { (pkg, _) ->
+            AttributionDocumentPdfModel(
+                purl = pkg.purl,
+                binaryFilename = pkg.binaryArtifact.takeUnless { it.url.isEmpty() }?.let { File(it.url).name },
+                declaredLicenses = createLicenseInfo(pkg.declaredLicensesProcessed, input.licenseTextProvider),
```

Comment:
Why do you only add declared licenses, does Antenna only operate on declared licenses by design?

Reply:
You are right, it should not only be the declared licenses, this should be changed.

Reply:
So, should i use all detected licenses instead of declared licenses?

Reply:
Not quite. If Antenna does not distinguish between declared and detected ("observed") licenses, then we should use the union of both.

Reply:
The idea is that the attribution document should contain the conclusion of detected license, declared license and further manual analysis in case of a complex or contradictive license situation. Isn‘t that the concluded license in ORT?

Reply:
Yes, but the concluded license is empty unless explicitly set (concluded) via ORT configuraton. Which means in the default case there would be no licenses, which is probably unexpected by the user.